### ISSUE TITLE
Make "Show in browser" work for alternative file browsers.

### ISSUE THREAD

**Issue by ian-r-rose** [jupyterlab/jupyterlab#4528]
#4500 added a nice new feature allowing the user to navigate the main file browser to the directory with the currently active document. This does not currently work for other filebrowsers (e.g., Google Drive, GitHub), but it could.

**Comment by telamonian** [comment]
Do document widgets or the context of a document widget have a property that would allow one to determine (or, better yet, fetch) the associated filebrowser? If so, I should be able to implement a fix for this pretty easily. If not, this issue may have to wait for #4529 to be fixed first.

**Comment by telamonian** [comment]
That's pretty simple. I'll get on that

**Comment by telamonian** [comment]
hmmm. I tried installing `@jupyterlab/github` and `@jupyterlab/google-drive` for dev/testing purposes, but I've run into some trouble. 

I installed them both following the instructions on their github pages. The extensions seem to run fine when dev mode is off, but when I run `jupyter lab --dev-mode` they vanish without a trace. I'm sure I just missed a section in the jupyterlab docs somewhere, but is there an extra step I have to take to enable extensions in dev mode?

**Comment by jasongrout** [comment]
You have to install them with "add:sibling": search for "add:sibling" in http://jupyterlab.readthedocs.io/en/stable/developer/extension_dev.html?highlight=add%3Asibling#extension-authoring

**Comment by telamonian** [comment]
Thanks for the info. Unfortunately though, I seem to have just made things worse. All of my git repos are stashed in a single `~/git` dir, so (based roughly on the discussion [here](https://github.com/jupyterlab/jupyterlab/issues/3522)) from inside my `jupyterlab` I ran

    jlpm run add:sibling ../jupyterlab-google-drive
    jlpm run add:sibling ../jupyterlab-github

but now I can't even build Jupyterlab. When I run `jlpm run build` I get a bunch of wacky errors:

```
../../node_modules/@types/text-encoding/index.d.ts(52,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'TextDecoder' must be of type '{ new (label?: string, options?: TextDecoderOptions): TextDecoder; prototype: TextDecoder; }', but here has type 'TextDecoderStatic'.
../../node_modules/@types/text-encoding/index.d.ts(54,13): error TS2403: Subsequent variable declarations must have the same type.  Variable 'TextEncoder' must be of type '{ new (): TextEncoder; prototype: TextEncoder; }', but here has type 'TextEncoderStatic'.
../jupyterlab-google-drive/src/drive/drive.ts(1223,34): error TS2345: Argument of type '(list: L) => Promise<any[]> | L[keyof L]' is not assignable to parameter of type '(value: L) => any[] | PromiseLike<any[]>'.
  Type 'Promise<any[]> | L[keyof L]' is not assignable to type 'any[] | PromiseLike<any[]>'.
    Type 'L[keyof L]' is not assignable to type 'any[] | PromiseLike<any[]>'.
      Type 'L[keyof L]' is not assignable to type 'PromiseLike<any[]>'.
../jupyterlab-google-drive/src/drive/drive.ts(1227,20): error TS2488: Type must have a '[Symbol.iterator]()' method that returns an iterator.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Doesn't seem worth dealing with, so I'm in the process of nuking my local Jupyterlab repo and starting over.

**Comment by telamonian** [comment]
Okay, so pretend I'm an idiot. What are the exact step-by-step commands that I have to run, given a fresh install/repo of Jupyterlab, in order to install dev mode compatible versions of `@jupyterlab/github` and `@jupyterlab/google-drive`?

**Comment by jasongrout** [comment]
We just upgraded to typescript 2.8, so you should delete the @types/text-encoding package (it's not compatible with ts 2.8)

**Comment by telamonian** [comment]
By "the package", you mean `@types/text-encoding`, right? I did indeed delete all trace of it. I suppose a fix will have to be pushed upstream to `@jupyterlab/github` as well

**Comment by jasongrout** [comment]
Yes, delete `@types/text-encoding` from `@jupyterlab/github` to get it to work with jlab master. You might also change the typescript version in that package to 2.8 too.

**Comment by ian-r-rose** [comment]
Thanks for chasing that down @telamonian, and for working out the `@types` issue! I know that it is sometimes tricky to get an extension built against master. And thanks for helping as well @jasongrout.

**Comment by ian-r-rose** [comment]
I can do a dev-mode install of the GitHub and Google Drive browsers for testing.

**Comment by jasongrout** [comment]
@ian-r-rose - was this fixed? It seems like a PR was merged fixing this.

**Comment by ian-r-rose** [comment]
Yep, thanks @telamonian!

**Comment by jasongrout** [comment]
Fixed in #4593.
